presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java

```diff
@@ -423,6 +441,12 @@ public PlanNode visitTableScan(TableScanNode tableScan, RewriteContext<Void> con
                     tableScan.getTableConstraints(),
                     pushdownFilterResult.getLayout().getPredicate(),
                     TupleDomain.all());
+
+            RowExpression unenforcedFilter = pushdownFilterResult.getUnenforcedConstraint();
+            if (!TRUE_CONSTANT.equals(unenforcedFilter)) {
```

Comment:
I think this should be good, but. I would maybe put it in a separate commit, since it's not related to the rest of the change.  That way if it causes an unexpected issue, we don't have to revert the rest of your changes.